src/ServiceControl.Monitoring/QueueLength/QueueLengthReportHandler.cs

```diff
@@ -15,7 +15,9 @@ public QueueLengthReportHandler(QueueLengthDataStore queueLengthDataStore)
 
         public Task Handle(MetricReport message, IMessageHandlerContext context)
         {
-            queueLengthDataStore.Store(message.Data);
+            var endpointInstanceId = EndpointInstanceId.From(context.MessageHeaders);
+
+            queueLengthDataStore.Store(endpointInstanceId.EndpointName, message.Data);
```

Comment:
Would it make sense to pass in the full endpoint instance id and have the Queue Length Data Store determine that it only cares about the name? That brings it in line with what the other stores do.